sky/policy.py

```diff
@@ -0,0 +1,154 @@
+"""Customize policy by users."""
+import copy
+import dataclasses
+import importlib
+import os
+import tempfile
+import typing
+from typing import Any, Callable, Dict, Optional
+
+from sky import dag as dag_lib
+from sky import sky_logging
+from sky import skypilot_config
+from sky.utils import common_utils
+from sky.utils import ux_utils
+
+logger = sky_logging.init_logger(__name__)
+
+if typing.TYPE_CHECKING:
+    from sky import task as task_lib
+
+
+@dataclasses.dataclass
+class UserTask:
+    task: 'task_lib.Task'
+    skypilot_config: Dict[str, Any]
+
+
+@dataclasses.dataclass
+class MutatedUserTask:
+    task: 'task_lib.Task'
+    skypilot_config: Dict[str, Any]
+
+
+class Policy:
```

Comment:
iface: This is confusing in the sense that the name suggests it's a iface to be implemented by users. E.g., it led me to think:
```
class AdminPolicy:
  def mutate(..., task) -> Task:  # to implement
  def admit(..., task) -> Optional[Exception]:  # to implement
```
but it's not. See Kyverno concepts. Shall we discuss changing the exposed interface to the above? At the very least, even if we don't change, we should rename the current class (or move its members out to become module funcs).

Reply:
Good point! I changed it to class, since it is easier to inherit and enforce the signature. I still keep a single function for it, since an admin can specify the rules in a single function.